server/jetty/src/main/java/io/deephaven/server/jetty/HomeFilter.java

```diff
@@ -16,8 +17,18 @@ public class HomeFilter implements Filter {
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
             throws IOException, ServletException {
-        if (response instanceof HttpServletResponse) {
-            ((HttpServletResponse) response).sendRedirect("/ide/");
+        if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
+            HttpServletRequest req = (HttpServletRequest) request;
+            HttpServletResponse resp = (HttpServletResponse) response;
+
+            final String location;
+            String queryString = req.getQueryString();
+            if (queryString != null) {
+                location = "/ide/?" + queryString;
```

Comment:
Assuming any encoding is already accounted for?

Reply:
Correct, getQueryString()'s contract is that it will not do any escaping, so it is already escaped. https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html#getQueryString--